Crash in Civilopedia

**Platform:** Desktop
**Version:** 4.3.5 (Build 771)
**Rulesets:** [Civ V - Vanilla, Civ V - Gods & Kings]
**Last Screen:** `com.unciv.ui.civilopedia.CivilopediaScreen`

--------------------------------

OS: Linux (amd64, 5.16.0-6-amd64)
	Debian GNU/Linux bookworm/sid
Java: Debian 11.0.14.1+1-post-Debian-1
	Max Memory: 3932 MB


--------------------------------


**Message:**
```
java.lang.RuntimeException: Actor: Group
|  SplitPane
|  |  Table
|  |  |  TextButton
|  |  |  |  Label: Fermer
|  |  |  AutoScrollPane
|  |  |  |  Table
|  |  |  
	at com.badlogic.gdx.scenes.scene2d.Actor.notify(Actor.java:192)
	at com.badlogic.gdx.scenes.scene2d.Actor.fire(Actor.java:152)
	at com.badlogic.gdx.scenes.scene2d.Stage.keyDown(Stage.java:418)
	at com.unciv.ui.UncivStage.access$keyDown$s80204510(UncivStage.kt:15)
	at com.unciv.ui.UncivStage$keyDown$1.invoke(UncivStage.kt:81)
	at com.unciv.ui.UncivStage$keyDown$1.invoke(UncivStage.kt:81)
	at com.unciv.ui.crashhandling.CrashHandlingExtensionsKt$wrapCrashHandling$1.invoke(CrashHandlingExtensions.kt:17)
	at com.unciv.ui.UncivStage.keyDown(UncivStage.kt:81)
	at com.badlogic.gdx.InputEventQueue.drain(InputEventQueue.java:58)
	at com.badlogic.gdx.backends.lwjgl3.DefaultLwjgl3Input.update(DefaultLwjgl3Input.java:189)
	at com.badlogic.gdx.backends.lwjgl3.Lwjgl3Window.update(Lwjgl3Window.java:378)
	at com.badlogic.gdx.backends.lwjgl3.Lwjgl3Application.loop(Lwjgl3Application.java:192)
	at com.badlogic.gdx.backends.lwjgl3.Lwjgl3Application.<init>(Lwjgl3Application.java:166)
	at com.unciv.app.desktop.DesktopLauncher.main(DesktopLauncher.kt:84)
Caused by: java.util.NoSuchElementException: Collection is empty.
	at kotlin.collections.CollectionsKt___CollectionsKt.first(_Collections.kt:201)
	at com.unciv.ui.civilopedia.CivilopediaScreen.navigateEntries(CivilopediaScreen.kt:314)
	at com.unciv.ui.civilopedia.CivilopediaScreen.access$navigateEntries(CivilopediaScreen.kt:37)
	at com.unciv.ui.civilopedia.CivilopediaScreen$9.invoke(CivilopediaScreen.kt:299)
	at com.unciv.ui.civilopedia.CivilopediaScreen$9.invoke(CivilopediaScreen.kt:299)
	at com.unciv.ui.utils.extensions.Scene2dExtensionsKt$installShortcutDispatcher$1.activate(Scene2dExtensions.kt:264)
	at com.unciv.ui.utils.extensions.Scene2dExtensionsKt$installShortcutDispatcher$1.keyDown(Scene2dExtensions.kt:230)
	at com.badlogic.gdx.scenes.scene2d.InputListener.handle(InputListener.java:53)
	at com.badlogic.gdx.scenes.scene2d.Actor.notify(Actor.java:188)
	... 13 more

```

**To Reproduce**

1. Open Civilopedia (this shows tutorials)
2. Press left key to see policies: a first bug here is that it shows nothing
3. Pressing up key crashes

--- a/core/src/com/unciv/ui/civilopedia/CivilopediaScreen.kt
+++ b/core/src/com/unciv/ui/civilopedia/CivilopediaScreen.kt
@@ -297,13 +297,17 @@ class CivilopediaScreen(
         globalShortcuts.add(Input.Keys.LEFT) {
             val categoryKey = categoryToEntries.keys
             val currentIndex = categoryKey.indexOf(currentCategory)
-            val targetCategory = categoryKey.elementAt((currentIndex - 1) % categoryKey.size)
+            val targetCategory = categoryKey.elementAt(
+                (currentIndex + categoryKey.size - 1) % categoryKey.size
+            )
             selectCategory(targetCategory)
         }
         globalShortcuts.add(Input.Keys.RIGHT) {
             val categoryKey = categoryToEntries.keys
             val currentIndex = categoryKey.indexOf(currentCategory)
-            val targetCategory = categoryKey.elementAt((currentIndex + 1) % categoryKey.size)
+            val targetCategory = categoryKey.elementAt(
+                (currentIndex + categoryKey.size + 1) % categoryKey.size
+            )
             selectCategory(targetCategory)
 
         }

--- a/core/src/com/unciv/ui/civilopedia/CivilopediaScreen.kt
+++ b/core/src/com/unciv/ui/civilopedia/CivilopediaScreen.kt
@@ -291,11 +291,22 @@ class CivilopediaScreen(
             else
                 selectEntry(link, noScrollAnimation = true)
 
-        for (categoryKey in CivilopediaCategories.values()) {
+        for (categoryKey in categoryToEntries.keys) {
             globalShortcuts.add(categoryKey.key) { navigateCategories(categoryKey.key) }
         }
-        globalShortcuts.add(Input.Keys.LEFT) { selectCategory(currentCategory.getByOffset(-1)) }
-        globalShortcuts.add(Input.Keys.RIGHT) { selectCategory(currentCategory.getByOffset(1)) }
+        globalShortcuts.add(Input.Keys.LEFT) {
+            val categoryKey = categoryToEntries.keys
+            val currentIndex = categoryKey.indexOf(currentCategory)
+            val targetCategory = categoryKey.elementAt((currentIndex - 1) % categoryKey.size)
+            selectCategory(targetCategory)
+        }
+        globalShortcuts.add(Input.Keys.RIGHT) {
+            val categoryKey = categoryToEntries.keys
+            val currentIndex = categoryKey.indexOf(currentCategory)
+            val targetCategory = categoryKey.elementAt((currentIndex + 1) % categoryKey.size)
+            selectCategory(targetCategory)
+
+        }
         globalShortcuts.add(Input.Keys.UP) { navigateEntries(-1) }
         globalShortcuts.add(Input.Keys.DOWN) { navigateEntries(1) }
         globalShortcuts.add(Input.Keys.PAGE_UP) { navigateEntries(-10) }

